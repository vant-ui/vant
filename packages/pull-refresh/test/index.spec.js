import PullRefresh from '..';
import { mount } from '@vue/test-utils';
import { triggerTouch, triggerDrag } from '../../../test/touch-utils';

test('change head content when pulling down', () => {
  const wrapper = mount(PullRefresh, {
    propsData: {
      value: false
    }
  });

  wrapper.vm.$on('input', value => {
    wrapper.vm.value = value;
  });

  const track = wrapper.find('.van-pull-refresh__track');

  // pulling
  triggerTouch(track, 'touchstart', 0, 0);
  triggerTouch(track, 'touchmove', 0, 10);
  expect(wrapper.html()).toMatchSnapshot();

  // loosing
  triggerTouch(track, 'touchmove', 0, 100);
  expect(wrapper.html()).toMatchSnapshot();

  // loading
  triggerTouch(track, 'touchend', 0, 100);
  expect(wrapper.html()).toMatchSnapshot();

  // still loading
  triggerDrag(track, 0, 100);
  expect(wrapper.html()).toMatchSnapshot();

  expect(wrapper.emitted('input')).toBeTruthy();
  expect(wrapper.emitted('refresh')).toBeTruthy();

  // end loading
  wrapper.vm.value = false;
  expect(wrapper.html()).toMatchSnapshot();
});

test('pull a short distance', () => {
  const wrapper = mount(PullRefresh, {
    propsData: {
      value: false
    }
  });

  const track = wrapper.find('.van-pull-refresh__track');
  triggerDrag(track, 0, 10);
  expect(wrapper.emitted('input')).toBeFalsy();
});

test('not in page top', () => {
  const wrapper = mount(PullRefresh, {
    propsData: {
      value: false
    }
  });

  window.scrollTop = 100;

  const track = wrapper.find('.van-pull-refresh__track');
  // ignore touch event when not at page top
  triggerDrag(track, 0, 100);
  window.scrollTop = 0;
  triggerTouch(track, 'touchmove', 0, 100);

  expect(wrapper.html()).toMatchSnapshot();
});
