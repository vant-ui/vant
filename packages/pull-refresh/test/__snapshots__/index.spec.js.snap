// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders pull-refresh correctly 1`] = `
<section class="van-doc-demo-section demo-pull-refresh" style="background:undefined;">
  <div class="van-pull-refresh">
    <div class="van-pull-refresh__track" style="transition:0ms;transform:translate3d(0,0px, 0);">
      <div class="van-pull-refresh__head">
        <!---->
        <!---->
        <!---->
      </div>
      <section class="van-doc-demo-block">
        <h2 class="van-doc-demo-block__title">基础用法</h2>
        <p>刷新次数: 0</p>
      </section>
    </div>
  </div>
</section>
`;
